### DISCUSSION TITLE
Node embedding training on large graph

### DISCUSSION THREAD

**Discussion by garyhsu29** [pyg-team/pytorch_geometric#5912] [Q&A] [answered]
Hello, I have a large dataset that contains around 200,000,000 nodes and 3,000,000,000 edges. In my case, it's a single large graph and the graph is dynamic. My goal is to learn node embedding for downstream classification task or use some clustering algorithms to find the community of nodes.

I have two questions here:
1. Because the graph is so large and I could not load the whole graph into my RAM, and I am wondering how to split my whole graph into multiple subgraphs without losing a lot of information about its topology.
2. What would be the suitable algorithm that I should use in my case? Do you think the graphsage + edge reconstruction loss could learn the embedding properly?

**Comment by rusty1s** [comment, marked as answer]
1. Wow, this is a giant graph. We are currently working on letting PyG scale to out-of-memory datasets via `FeatureStore` and `GraphStore` abstractions (will be part of the next release). Currently, you would need to rely on external sources to split your graph before inputting into PyG.

2. Yes, that sounds good. `GraphSAGE` is always a strong baseline.

**Reply by garyhsu29** [reply]
Thank you so much!
